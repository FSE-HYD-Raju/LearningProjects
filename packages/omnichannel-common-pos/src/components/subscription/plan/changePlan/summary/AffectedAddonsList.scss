.AffectedAddonsList {
	&-title-container {
		display: flex;
		flex-direction: row;
	}
	&-addon-icon {
		transform: translateY(0.125rem);
		font-size: 1.125rem;
	}
	&-title {
		padding: 0 0 0 0.125rem;
		b {
			font-weight: 500;
		}
	}
	&-addons-list {
		display: flex;
		flex-direction: column;
		padding: 0 0 0 1.125rem;
	}
	&-addons-list-item,
	&-view-all-button,
	&-show-less-button {
		padding: 0.5rem 0 0 0;
	}
	@include media-breakpoint-up(sm) {
		&-addon-icon {
			transform: translateX(-1.375rem);
		}
		&-title {
			display: inline-block;
			transform: translateX(-1.125rem);
		}
		&-addons-list {
			padding: unset;
		}
	}
}
