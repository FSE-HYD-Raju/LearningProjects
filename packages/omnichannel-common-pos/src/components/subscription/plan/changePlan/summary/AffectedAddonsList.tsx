import * as R from "react";
import cssns from "../../../../../utils/cssnsConfig";
import FormattedMessage from "../../../../../channelUtils/FormattedMessage";
const React = cssns("AffectedAddonsList").React as typeof R;
import { FormattedHTMLMessage } from "../../../../../channelUtils";
import { FormattedMessage as IntlFormattedMessage } from "react-intl";
import messages from "./ChangePlanSummary.messages";

interface AffectedAddonsListProps {
	titleMessage: IntlFormattedMessage.MessageDescriptor;
	addonsNames: string[];
	maxInitialCountToShow: number;
	iconClassName: string;
	className?: string;
}
interface AffectedAddonsListState {
	isExpanded: boolean;
}
class AffectedAddonsList extends React.Component<AffectedAddonsListProps, AffectedAddonsListState> {
	constructor(props: AffectedAddonsListProps) {
		super(props);
		this.state = { isExpanded: false };
	}
	expand = () => {
		this.setState({ isExpanded: true });
	};
	collapse = () => {
		this.setState({ isExpanded: false });
	};
	render() {
		const { addonsNames, titleMessage, maxInitialCountToShow, iconClassName, className } = this.props;
		const { isExpanded } = this.state;
		const isViewMoreVisible = !isExpanded && addonsNames.length > maxInitialCountToShow;
		const isShowLessVisible = isExpanded;
		const addonsToShow = isExpanded ? addonsNames : addonsNames.slice(0, maxInitialCountToShow);

		return addonsNames.length > 0 ? (
			<div className={`this ${className}`}>
				<div className="title-container">
					<i className={`fa ${iconClassName} addon-icon`} />
					<div className="title">
						<FormattedHTMLMessage {...titleMessage} values={{ count: addonsNames.length }} />
					</div>
				</div>
				<div className="addons-list">
					{addonsToShow.map((addonName: string, index: number) => (
						<span className="addons-list-item" key={index}>
							{addonName}
						</span>
					))}
					{isViewMoreVisible && (
						<a className="btn-link view-all-button" onClick={this.expand} data-test-name="view-all-button" href="javascript:">
							<FormattedMessage {...messages.viewAllAddonsButton} />
						</a>
					)}
					{isShowLessVisible && (
						<a className="btn-link show-less-button" onClick={this.collapse} data-test-name="show-less-button" href="javascript:">
							<FormattedMessage {...messages.showLessAddonsButton} />
						</a>
					)}
				</div>
			</div>
		) : null;
	}
}
export { AffectedAddonsList, AffectedAddonsListProps };
