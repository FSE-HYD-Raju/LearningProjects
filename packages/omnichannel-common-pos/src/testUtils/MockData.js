import _ from "lodash";
import R from "ramda";

const firstDateThisWeek = () => {
	const date = new Date();
	while (date.getDay() !== 0) {
		date.setDate(date.getDate() - 1);
	}
	return date;
};

const people = [
	{
		firstName: "Kathrin",
		lastName: "K",
		gender: "female",
		language: "swe",
		countryOfBirth: "SE",
		birthDay: firstDateThisWeek(),
		emails: [{ email: "kathrin.k@qvantel.com" }],
		email: "kathrin.k@qvantel.com",
		mobileNumbers: [{ number: "2398457" }],
		mobileNumber: "2398457",
		fixedLineNumbers: [{ number: "2398457" }],
		fixedLineNumber: "2398457",
		password: "qwerasdf",
		passwordConfirmation: "qwerasdf"
	},
	{
		firstName: "John",
		lastName: "Smith",
		gender: "male",
		language: "fin",
		birthDay: firstDateThisWeek(),
		countryOfBirth: "United Kingdom",
		emails: [{ email: "john.smith@qvantel.com" }],
		email: "john.smith@qvantel.com",
		mobileNumbers: [{ number: "3498573" }],
		mobileNumber: "3498573",
		fixedLineNumbers: [{ number: "52497584" }],
		fixedLineNumber: "52497584"
	}
];

const basketItems = [
	{
		id: "7879c247-9e98-40ee-bb48-544eb366494a",
		type: "basketItems",
		attributes: {
			product: {
				name: "Movie Steaming Service"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 1.9,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 14.9,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: [],
			totalUpfrontPrice: 1.9,
			totalUpfrontPriceWithChildren: 1.9
		}
	},
	{
		id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
		type: "basketItems",
		attributes: {
			product: {
				name: "XDSL"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 100.6,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 30,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: []
		}
	}
];

const deliveryItem = {
	id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
	type: "basketItems",
	inputtedCharacteristics: {
		CH_Delivery_Contact_Info: "5c5dddf6-72be-4d0a-9711-6027f247af28"
	},
	product: {
		name: "Home delivery",
		id: "PO_Home_Delivery"
	}
};

export default class MockData {
	static getPerson(searchCriteria) {
		if (searchCriteria) {
			return _.find(people, searchCriteria);
		} else {
			return people[0];
		}
	}

	static getPeople() {
		return people;
	}

	static getPostalAddress() {
		return {
			street: "1 Test St",
			coAddress: "Best Ltd.",
			postalCode: "90210",
			city: "Test City",
			country: "FI"
		};
	}

	static getUser() {
		return {
			"id": "bf588945-ce95-490a-a774-8ef91661a146",
			"type": "persons",
			"attributes": {
				"lastName": "Testeri",
				"addresses": null,
				"gender": "male",
				"fixedLineNumbers": [],
				"formattedName": "Teppo Testeri",
				"roles": null,
				"billingAccountId": null,
				"language": null,
				"emails": [
					{
						"role": "PRIMARY",
						"id": "2ab79d7b-862d-49df-8afc-e6104a914919",
						"email": "tepsteri@tepinposti.fi"
					}
				],
				"honorificSuffix": null,
				"countryOfBirth": "FI",
				"personRelationships": [],
				"birthDay": null,
				"placeOfBirth": null,
				"characteristics": null,
				"mobileNumbers": [
					{
						"role": "PRIMARY",
						"id": "a843f46d-f729-40dd-a7ce-240a64bc008e",
						"number": "1234576890"
					}
				],
				"honorificPrefix": null,
				"avatar": "/img/icon-user-male.svg",
				"privacySettings": {
					"privacy1": false,
					"privacy2": false
				},
				"postalAddresses": [
					{
						"role": "PRIMARY",
						"id": "5e30221d-ad19-4043-8088-f594f5fdf77f",
						"street": "Piippukatu 11 A 1",
						"coAddress": null,
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": null,
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					},
					{
						"role": "DELIVERY",
						"id": "5c5dddf6-72be-4d0a-9711-6027f247af28",
						"street": "Piippukatu 11 A 1",
						"coAddress": "Piippukatu 11 A 1",
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": "",
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					}
				],
				"identifications": [],
				"firstName": "Teppo",
				"customerAccountId": null,
				"nationality": null,
				"additionalName": null,
				"maritalStatus": null
			}
		}
	}

	static getBasket(withItems: boolean, withDeliveryItems: boolean) {
		const basket = {
			activeBasket: {
				attributes: {
					totalPrices: [],
					totalUpfrontPrice: 0
				}
			}
		};

		if (withItems) {
			basket.basketItems = basketItems;

			const totalPriceOneTime = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);
			const totalPriceMonthly = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);

			basket.activeBasket.attributes.totalPrices = [
				{
					type: "ONE_TIME",
					taxFreeAmount: totalPriceOneTime,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: totalPriceMonthly,
					taxRate: 0,
					currency: "EUR"
				}
			];

			if (withDeliveryItems) {
				basket.basketItems[0].attributes.childBasketItems.push(
					deliveryItem
				);
			}
		}

		return basket;
	}
}
