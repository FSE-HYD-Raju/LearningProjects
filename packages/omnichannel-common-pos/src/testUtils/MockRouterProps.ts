import {
	RouteComponentProps,

} from "react-router";

import {
	Href,
	UnregisterCallback,
	TransitionPromptHook,
	LocationState,
	Path,
	LocationDescriptorObject,
	LocationListener
} from "history";

const mockLocation: RouteComponentProps<any>["location"] = {
		hash: "",
		key: "",
		pathname: "",
		search: "",
		state: {}
};

const mockHistory: RouteComponentProps<any>["history"] = {
	length: 2,
	action: "POP",
	location: mockLocation,
	push: (path: Path | LocationDescriptorObject, state?: LocationState): void => {},
	replace: (path: Path | LocationDescriptorObject, state?: LocationState): void => {},
	go: (n: number): void => {},
	goBack: (): void => {},
	goForward: (): void => {},
	block: (prompt?: boolean | string | TransitionPromptHook): UnregisterCallback => { return () => {}; },
	listen: (listener: LocationListener): UnregisterCallback => {return () => {}; },
	createHref: (location: LocationDescriptorObject): Href => { return ""; },
};

const mockRouterProps: RouteComponentProps<any> = {
		match: {
			isExact: true,
			params: {},
			path: "",
			url: ""
		},
		location: mockLocation,
		history: mockHistory,
		staticContext: {}
};

export {
	mockRouterProps,
	mockLocation,
	mockHistory,
	RouteComponentProps
};
