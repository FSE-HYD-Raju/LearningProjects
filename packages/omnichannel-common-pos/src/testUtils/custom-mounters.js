import React from "react";
import { IntlProvider, intlShape } from "react-intl";
import { mount, shallow } from "enzyme";
import PropTypes from "prop-types";
import R from "ramda";
import TestUtils from "./TestUtils";

const MAX_DIVE_DEEP_LEVEL = 16;
const intlProvider = new IntlProvider({ locale: "en" }, {});
const { intl } = intlProvider.getChildContext();

const baseContext = {
	flux: {
		persist: () => {}
	},
	store: TestUtils.mockReduxStore(),
	intl,
	location: {},
	only: {},
	skip: {}
};

const nodeWithIntlProp = (node, mockedRouterOptions) => {
	if (mockedRouterOptions) {
		mockedRouterOptions.props().history.push = jest.fn();
	}
	return React.cloneElement(node, {
		intl,
		...(mockedRouterOptions ? mockedRouterOptions.props() : {})
	});
};

function getOptionsForShallowMount(options, debug, mockedRouterContextOptions) {
	const mountOptions = R.merge(options ? options : {}, {
		disableLifecycleMethods: !(
			options && options.disableLifecycleMethods === false
		),
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				intl: intlShape,
				store: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mountOptions;
}
function deepDive(wrapper) {
	let diveAttemptsCount = 0;
	while (
		wrapper.hostNodes().length === 0 &&
		diveAttemptsCount < MAX_DIVE_DEEP_LEVEL
	) {
		wrapper = wrapper.dive();
		diveAttemptsCount += 1;
	}
	return wrapper;
}
export function shallowWithContextAndRouterProps(node, options, debug) {
	return shallowWithContext(node, options, debug, true);
}
export function shallowWithContextAndLifecycle(node, options, debug) {
	return shallowWithContext(
		node,
		R.merge(options || {}, { disableLifecycleMethods: false }),
		debug,
		true
	);
}
export function deepShallowWithContextAndLifecycle(node, options, debug) {
	return deepDive(shallowWithContextAndLifecycle(node, options, debug));
}
export function deepShallowWithContext(node, options, debug) {
	return deepDive(shallowWithContext(node, options, debug, true));
}
export function shallowWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	return shallow(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		getOptionsForShallowMount(options, debug, mockedRouterContextOptions)
	);
}

export function mountWithContextAndRouterProps(node, options, debug) {
	return mountWithContext(node, options, debug, true);
}
export function mountWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	const mountOptions = R.merge(options ? options : {}, {
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				flux: PropTypes.object,
				store: PropTypes.object,
				intl: intlShape,
				location: PropTypes.object,
				only: PropTypes.object,
				skip: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mount(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		mountOptions
	);
}

const attachmentElementId = "__OC_COMPONENT_TEST_ATTACHMENT_ELEMENT__";

export function attachWithContext(node, options) {
	const attachmentElement = document.createElement("div");
	attachmentElement.setAttribute("id", attachmentElementId);
	document.body.appendChild(attachmentElement);

	const wrapper = mountWithContext(
		node,
		R.merge(options, { attachTo: attachmentElement })
	);
	const doppelWrapper = hijackDetacher(wrapper);

	return doppelWrapper;
}

function hijackDetacher(wrapper) {
	const doppel = wrapper;
	doppel.__detach = wrapper.detach;

	doppel.detach = () => {
		removeAttachmentElements();
		doppel.__detach.apply(wrapper);
	};

	return doppel;
}

function removeAttachmentElements() {
	document.body.querySelectorAll("#" + attachmentElementId).forEach(elem => {
		document.body.removeChild(elem);
	});
}

export function detach(wrapper) {
	removeAttachmentElements();
	wrapper.detach();
}
