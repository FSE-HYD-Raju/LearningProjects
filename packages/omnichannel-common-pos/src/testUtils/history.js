import isClient from "../utils/isClient";
import { createBrowserHistory, createMemoryHistory } from "history";
import { useRouterHistory } from "react-router";

const history = isClient
	? useRouterHistory(createBrowserHistory)({ queryKey: false })
	: createMemoryHistory();

export default history;
