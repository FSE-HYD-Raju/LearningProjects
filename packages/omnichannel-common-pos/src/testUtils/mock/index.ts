import { default as productOffering } from "./ProductOfferingMock";
import { ProductMock as product } from "./ProductMock";
import { AgreementMock as agreement } from "./AgreementMock";
import { ChargingBalancesMock as chargingBalances } from "./ChargingBalancesMock";
import { default as postalAddress } from "./PostalAddressMock";
import { default as identification } from "./IdentificationMock";
import { default as user } from "./UserMock";
import { default as order } from "./OrderMock";
import { default as basket } from "./BasketMock";
export { productOffering, postalAddress, identification, user, order, basket, product, agreement, chargingBalances };
