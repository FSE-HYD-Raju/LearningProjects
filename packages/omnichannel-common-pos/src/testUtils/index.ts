import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
