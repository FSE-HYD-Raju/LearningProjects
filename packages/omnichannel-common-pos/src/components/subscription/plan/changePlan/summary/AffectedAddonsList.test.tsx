import * as React from "react";
import { AffectedAddonsList, AffectedAddonsListProps } from "./AffectedAddonsList";
import { shallowWithContext, mountWithContext } from "../../../../../testUtils";

describe("AffectedAddonsList", () => {
	const minProps: AffectedAddonsListProps = {
		titleMessage: {
			id: "mid",
			defaultMessage: "here,{count},addons",
		},
		iconClassName: "fa-test",
		maxInitialCountToShow: 3,
		addonsNames: ["one", "two"],
		className: "custom",
	};

	it("snapshot with minimal props", () => {
		const wrapper = shallowWithContext(<AffectedAddonsList {...minProps} />);
		expect(wrapper).toMatchSnapshot();
	});
	it("should not render anything when no addons", () => {
		const wrapper = mountWithContext(<AffectedAddonsList {...minProps} addonsNames={[]} />);
		expect(wrapper.isEmptyRender()).toBeTruthy();
	});
	it("should not render 'view all' button when addons count is less than max", () => {
		const wrapper = mountWithContext(<AffectedAddonsList {...minProps} maxInitialCountToShow={10} />);
		expect(wrapper.find('[data-test-name="view-all-button"]').exists()).toBeFalsy();
	});
	it("should expand and collapse after clicking 'view all'/'show less' buttons", () => {
		const wrapper = mountWithContext(<AffectedAddonsList {...minProps} maxInitialCountToShow={2} addonsNames={["one", "two", "three"]} />);
		const viewAllButton = wrapper.find('[data-test-name="view-all-button"]');

		expect(wrapper.text()).not.toContain("three");
		expect(wrapper.find('[data-test-name="show-less-button"]').exists()).toBeFalsy();

		viewAllButton.simulate("click");

		expect(wrapper.find('[data-test-name="view-all-button"]').exists()).toBeFalsy();
		expect(wrapper.text()).toContain("three");

		const showLessButton = wrapper.find('[data-test-name="show-less-button"]');
		showLessButton.simulate("click");

		expect(wrapper.text()).not.toContain("three");
		expect(wrapper.find('[data-test-name="view-all-button"]').exists()).toBeTruthy();
		expect(wrapper.find('[data-test-name="show-less-button"]').exists()).toBeFalsy();
	});
});
