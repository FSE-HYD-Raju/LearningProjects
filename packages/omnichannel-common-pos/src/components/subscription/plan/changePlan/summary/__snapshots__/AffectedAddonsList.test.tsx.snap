// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AffectedAddonsList snapshot with minimal props 1`] = `
<div
  className="AffectedAddonsList AffectedAddonsList-custom"
>
  <div
    className="AffectedAddonsList-title-container"
  >
    <i
      className="fa fa-test AffectedAddonsList-addon-icon"
    />
    <div
      className="AffectedAddonsList-title"
    >
      <FormattedHTMLMessageWrapper
        defaultMessage="here,{count},addons"
        id="mid"
        values={
          Object {
            "count": 2,
          }
        }
        wrapperType="wrapper"
      />
    </div>
  </div>
  <div
    className="AffectedAddonsList-addons-list"
  >
    <span
      className="AffectedAddonsList-addons-list-item"
      key="0"
    >
      one
    </span>
    <span
      className="AffectedAddonsList-addons-list-item"
      key="1"
    >
      two
    </span>
  </div>
</div>
`;
